Content/Items/Jungle/Weapons.HauntedDaggerStaff.cs

```diff
@@ -0,0 +1,509 @@
+						Main.rand.NextVector2Circular(2f, 2f), 0, new Color(70, 200, 100), 0.45f);
+				}
+
+				return false;
+			}
+
+			return true;
+		}
+
+		public override void AI()
+		{
+			UpdateProjectile();
+
+			if (AttackDelay > 0)
+				AttackDelay--;
+
+			if (rotTimer > 0)
+				rotTimer--;
+
+			if (MinionTarget != null && AttackDelay <= 0)
+				TargetWhoAmI = MinionTarget.whoAmI;
+
+			if (!FoundTarget)
+			{
+				if (Main.rand.NextBool(60))
+				{
+					Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2Circular(2f, 2f), ModContent.DustType<Dusts.GlowFastDecelerate>(),
+						Main.rand.NextVector2Circular(2f, 2f), 0, new Color(70, 200, 100), 0.45f);
+				}
+
+				DoIdleMovement();
+
+				if (AttackDelay <= 0)
+				{
+					NPC target = FindTarget();
+					if (target != default)
+						TargetWhoAmI = target.whoAmI;
+				}			 
+			}
+			else if (AttackDelay <= 0)
+			{
+				attackTimer++;
+
+				if (attackTimer < 45)
+				{
+					DoIdleMovement();
+
+					rotTimer += (int)MathHelper.Lerp(1f, 50f, EaseBuilder.EaseCubicIn.Ease(attackTimer / 45f));
+
+					float lerper = MathHelper.Lerp(30f, 2f, EaseBuilder.EaseCubicIn.Ease(attackTimer / 45f));
+					if (Main.rand.NextBool(3))
+						Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2CircularEdge(lerper, lerper), ModContent.DustType<Dusts.GlowFastDecelerate>(), Main.rand.NextVector2Circular(0.25f, 0.25f), 0, new Color(70, 200, 100), 0.5f);
+				}
+				else
+				{
+					if (Main.rand.NextBool())
+					{
+						Dust.NewDustPerfect(Projectile.Center + Main.rand.NextVector2Circular(2f, 2f), ModContent.DustType<Dusts.GlowFastDecelerate>(),
+							Main.rand.NextVector2Circular(2f, 2f), 0, new Color(70, 200, 100), 0.45f);
+					}
+
+					rotationalVelocity = Vector2.Lerp(rotationalVelocity, Projectile.DirectionTo(Target.Center), 0.15f);
+					Projectile.rotation = rotationalVelocity.ToRotation() + MathHelper.PiOver2;
+
+					if (attackTimer == 45)
+					{
+						Projectile.velocity += Projectile.DirectionTo(Target.Center) * 22f;
+
+						Helpers.Helper.PlayPitched("Effects/HeavyWhooshShort", 0.65f, Main.rand.NextFloat(-0.1f, 0.1f), Projectile.Center);
+					}
+
+					Vector2 direction = Target.Center - Projectile.Center;
+					direction.Normalize();
+					direction *= 23.5f;
+					Projectile.velocity = Vector2.Lerp(Projectile.velocity, direction, 0.075f);
+
+					if (rotTimer > 0)
+						rotTimer = 0;
+				}
+
+				if (!Target.active || Target.Distance(Owner.Center) > 1000f)
+				{
+					TargetWhoAmI = -1;
+					AttackDelay = maxAttackDelay;
+					attackTimer = 0;
+				}
+			}
+		}
+
+		public override void OnHitNPC(NPC target, NPC.HitInfo hit, int damageDone)
+		{
+			if (!Embedded && target.life > 0)
+			{
+				EmbeddedTarget = target;
+				Projectile.ai[0] = 1f;
+				Projectile.friendly = false;
+				Projectile.tileCollide = false;
+				enemyOffset = Projectile.position - target.position;
+				enemyOffset -= Projectile.velocity;
+				Projectile.netUpdate = true;
```

Comment:
I would recommend setting shouldSendHitPacket to true in the owner StarlightPlayer for hit syncing instead so it gets all the dusts and sounds too.

Reply:
actually that may not be sufficient here since this is the proj itself doing the logic and not actually using our hooks